### PR TITLE
refactor: Refactor `SurfaceReached` aborter

### PR DESCRIPTION
**Pull request by andiwand**
With these changes I try to decouple the navigator and surface reached aborter further. I don't believe that the navigator has to care about the target surface other than optimization reasons while the surface reached aborter should only care about the surface it is supposed to care about without looking at or touching the navigation state.

blocked by
- https://github.com/acts-project/acts/pull/2621

### CODE REVIEW

**Comment by andiwand** [comment]
there is some physmon diff for the gsf but it looks marginal to me. maybe @benjaminhuth can have a look. will not update the reference for now

**Comment by benjaminhuth** [comment]
Seems to improve things actually, right?

Maybe because the overstep limit is now properly propagated?

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp [hunk 1]
```diff
@@ -345,11 +345,12 @@ class CombinatorialKalmanFilter {
...
 
-    /// The filter target surface
-    const Surface* filterTargetSurface = nullptr;
+    /// The filter target surface aborter
+    SurfaceReached filterTargetReached{std::numeric_limits<double>::lowest()};
 
-    /// The smoothing target surface
-    const Surface* smoothingTargetSurface = nullptr;
+    /// The smoothing target surface aborter
+    SurfaceReached smoothingTargetReached{
```
Comment (paulgessinger) [inline Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:352]: What if we disable the smoothing? I mentioned that we probably have to externalize the CKF smoothing to be able to copy them out. If we turn off the smoothing, will it also not try to reach this surface?

**Review by paulgessinger — COMMENTED** [review]
I think i like this.

File: Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp (hunk 1 above)
Comment (andiwand) [reply Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp:352]: this should still do the same thing if not set, yes

File: Core/include/Acts/Propagator/MultiStepperAborters.hpp [hunk 2]
```diff
@@ -99,14 +70,23 @@ struct MultiStepperSurfaceReached {
...
+
+      if (!SurfaceReached::operator()(singleState, singleStepper, navigator,
+                                      logger)) {
+        reached = false;
+      } else {
+        cmp.status() = Acts::Intersection3D::Status::onSurface;
+      }
     }
 
     return reached;
```
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/MultiStepperAborters.hpp:88]: Maybe we should add VERBOSE log messages like above for this case? (like the "maybe next time?" log message below)

**Review by benjaminhuth — COMMENTED** [review]
Looks good from my side! (mainly looked at the multi stepper aborter)

File: Core/include/Acts/Propagator/MultiStepperAborters.hpp (hunk 2 above)
Comment (benjaminhuth) [reply Core/include/Acts/Propagator/MultiStepperAborters.hpp:88]: Ah actually I now see that we get a bunch of logging messages when calling the parent `operator()`...

But I thought a closing VERBOSE message reporting the status of `reached` wouldn't hurt right?

**Review by benjaminhuth — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [eae867a3e3900141078116741847cd8166eeb916]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093868) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093867) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093866) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093865) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093864) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093863) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093862) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093861) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093860) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093859) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093858) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093857) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093856) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093855) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093854) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093853) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093852) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093851) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093850) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34093849) |

**Pull request merged**